applications/nrf_desktop/configuration/nrf52840_pca20041/selector_hw_def.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "pins.h"
+
+/* This configuration file is included only once from selector_hw module
+ * and holds information about pins used by module.
+ */
+
+/* This structure enforces the header file is included only once in the build.
+ * Violating this requirement triggers a multiple definition error at link time.
+ */
+const struct {} selector_hw_def_include_once;
+
+static const char * const port_map[] = {
+	DT_GPIO_P1_DEV_NAME
+};
+
+static const struct pin pins[][CONFIG_DESKTOP_SELECTOR_HW_MAX_STATES] = {
```

Comment:
This is compled only when `CONFIG_SELECTOR_HW_ENABLE `is chosen. Is it possible that `CONFIG_DESKTOP_SELECTOR_HW_MAX_STATES `will be undefined then?

Reply:
If you forget to put CONFIG_DESKTOP_SELECTOR_HW_MAX_STATES  then it is possible. After that it ends with compilation error instead of meaningful message provided by you.

Reply:
I checked and it simply has value 1 assigned (and it compiles without issues).

Reply:
Since standard does not state that it is always properly defined in such situation, I will add compilation error for this one.